headers/types/ground_mode/ground_mode.h

```diff
@@ -5,6 +5,36 @@
 
 #include "enums.h"
 
+// Scripting coroutine located in unionall
+struct script_coroutine {
+    // 0x0: Offset (in halfwords) where the coroutine starts, relative to the start of unionall
+    uint16_t offset;
+    uint16_t type;      // 0x2: Not confirmed
+    uint16_t linked_to; // 0x4: From SkyTemple's source code. Purpose unknown.
+};
+ASSERT_SIZE(struct script_coroutine, 6);
```

Comment:
I'm curious if this is the same struct as the entries in `C_ROUTINES`, which are 8 bytes wide? I'd love to finally give that global a type.

Reply:
Are you sure `common_routine` is 8 bytes long? Because `script_coroutine` is definitely 6 bytes, the ASM addresses the list as `corutine_id * 6`.

Reply:
Oh wait I forgot I already defined a type for it lol. Yes, I'm sure that one's 8 bytes wide. You can look at the `C_ROUTINES` table to confirm; it's pretty easy to spot since you have a string pointer every 8 bytes.

I suppose `struct script_coroutine` is the struct for a "loaded" coroutine (since it has a memory offset, mentions of linking), whereas `struct common_routine` is just part of the static table (ID and name)? I wonder if we should put them next to each other and add comments detailing the difference.

Reply:
Looks like a good distinction to make. I'll change it.